pkgs/development/compilers/oraclejdk/jdk-linux-base.nix

```diff
@@ -36,24 +37,33 @@
 
 assert swingSupport -> xorg != null;
 
+if !licenseAccepted then throw ''
+    You must accept the Oracle Binary Code License Agreement for Java SE at
+    https://www.oracle.com/technetwork/java/javase/terms/license/index.html
+    by setting nixpkgs config option 'oraclejdk.accept_license = true;'
+  ''
+else assert licenseAccepted;
+
 let
 
   /**
    * The JRE libraries are in directories that depend on the CPU.
    */
-  architecture = {
+  architectures = {
     i686-linux    = "i386";
     x86_64-linux  = "amd64";
     armv7l-linux  = "arm";
     aarch64-linux = "aarch64";
-  }.${stdenv.hostPlatform.system};
+  };
+
+  architecture = architectures.${stdenv.hostPlatform.system};
```

Comment:
I do admit `architecture` is a bad name, but adding `architectures` too makes this more confusing. maybe something like `architecturesMap`, if we really need it (see below)?